### PR TITLE
Improve code style and update dependencies

### PR DESCRIPTION
**Pull request by vaihtovirta**
- introduce eslint
- apply eslint rules
- bump react, react-dom, react-hot-loader

### CODE REVIEW

File: src/popover.jsx [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
+
+export default class Popover extends Component {
+  static propTypes = {
+    className: PropTypes.string,
+    closeOnOuterClick: PropTypes.bool,
+    horizontalJustify: PropTypes.oneOf(["left", "right"]),
+    isOpen: PropTypes.bool,
+    leftOffset: React.PropTypes.number,
+    position: PropTypes.oneOf(["top", "bottom", "left", "right"]),
+    toggleButton: PropTypes.node.isRequired,
```
Comment (dphaener) [inline src/popover.jsx]: This button prop is not required. Down below we're checking for the presence of it and rendering a default button if it is not provided.

**Comment by nmn** [comment]
I like what this PR does. Though I would recommend using an eslint-config like `standard` or `airBnB` to avoid managing the long list of rules.

**Comment by vaihtovirta** [comment]
@dphaener @nmn Hey guys, I've applied your comments, added `eslint-config-airbnb` and rebased onto 1.x branch. 
Please take a look and merge if it's ok. Thanks.

**Comment by dphaener** [comment]
@vaihtovirta Looks great! I'll go ahead and merge. I made some changes to master so we'll need to rebase the changes from master, add a few tests in critical places, then I think we'll be ready for a 1.0 release! 

Thanks for the work on this!

**Pull request merged**